archive_viewer/table_models/curve_model.py

```diff
@@ -206,11 +238,38 @@ def set_model_curves(self, curves: List[Dict] = []) -> None:
             for k, v in c.items():
                 if v is None:
                     del c[k]
-            c['y_channel'] = c['channel']
-            del c['channel']
-            self._plot.addYChannel(**c)
-            self._row_names.append(self.next_header())
-        self.append(addAxis=False)
+            if 'channel' in c:
+                c['y_channel'] = c['channel']
+                del c['channel']
+                self._plot.addYChannel(**c)
+                self._row_names.append(self.next_header())
+            else:
+                self.append()
+                headers = re.findall("{(.+?)}", c['formula'])
+                splitFormula = re.split("{.+?}", c['formula'])
+                newFormula = splitFormula.pop(0)
+                for header in headers:
+                    # For every header refereced in our formula
+                    if header in c['curveDict'].keys():
+                        # If this curve (formula or channel) is in the curves list
+                        name = c['curveDict'][header]
+                        if name in self:
+                            # Manually find it and replace the header in the formula
+                            for i in range(len(self.plot.curves)):
+                                curve = self._plot._curves[i]
+                                if ((hasattr(curve, "address") and curve.address == name)
+                                            or (hasattr(curve, "formula") and curve.formula == name)):
+                                    newFormula += "{" + self._row_names[i] + "}" + splitFormula.pop(0)
+
+                                    break
```

Comment:
Styling: You can make this indentation look a bit neater by checking for the opposite conditions and ending the iteration early if they're true. This should be a bit easier to read while performing the same function:
```suggestion
                for header in headers:
                    # For every header referenced in our formula
                    if header not in c['curveDict'].keys():
                        continue

                    # If this curve (formula or channel) is in the curves list
                    name = c['curveDict'][header]
                    if name not in self:
                        continue

                    # Manually find it and replace the header in the formula
                    for i in range(len(self.plot.curves)):
                        curve = self._plot._curves[i]
                        found_curve = hasattr(curve, "address") and curve.address == name
                        found_curve |= hasattr(curve, "formula") and curve.formula == name

                        if found_curve:
                            newFormula += "{" + self._row_names[i] + "}" + splitFormula.pop(0)
                            break
```

Reply:
Sorry this change is so big btw. It's a large block of code